Testing/MeshTests/MeshTests.cpp

```diff
@@ -14,3 +18,53 @@ TEST(MeshTests, coverage_test) {
   ASSERT_TRUE(pelvis.compare_scalars_equal(baseline));
   ASSERT_TRUE(true);
 }
+
+
+TEST(MeshTests, warpTest1)
+{
+  Mesh ellipsoid( std::string(TEST_DATA_DIR) + "/ellipsoid_0.ply");
+  Mesh::MeshType inMesh = ellipsoid.get_poly_data();
+  Eigen::MatrixXd Vref = MeshUtils::distilVertexInfo(inMesh);
+  Eigen::MatrixXi Fref = MeshUtils::distilFaceInfo(inMesh);
+  std::string particlePath = std::string(TEST_DATA_DIR) + "/ellipsoid_0.particles";
+  std::vector<std::string> paths;
+  paths.push_back(particlePath);
+  ParticleSystem particlesystem(paths);
+  Eigen::MatrixXd staticPts = particlesystem.Particles();
+  staticPts.resize(3, 128);
+  Eigen::MatrixXd W1 = MeshUtils::generateWarpMatrix(Vref, Fref, staticPts.transpose());
+  Mesh output = MeshUtils::warpMesh(staticPts.transpose(), W1, Fref);
+  Mesh::MeshType inMesh2 = output.get_poly_data();
+  Eigen::MatrixXd Vref2 = MeshUtils::distilVertexInfo(inMesh2);
+  Eigen::MatrixXi Fref2 = MeshUtils::distilFaceInfo(inMesh2);
+  Eigen::VectorXi J = Eigen::VectorXi::LinSpaced(Vref2.rows(),0,Vref2.rows()-1);
+	Eigen::VectorXd sqrD;
+	Eigen::MatrixXd _2;
+  Eigen::VectorXi b;
+	igl::point_mesh_squared_distance(staticPts.transpose(),Vref2,J,sqrD,b,_2);
+  bool chkdst = (!(sqrD.sum() > 1.0e-10)) && (!(sqrD.maxCoeff() > 1.0e-10));
+  ASSERT_TRUE(chkdst);
+}
+
+TEST(MeshTests, warpTest2)
+{
+  Mesh ellipsoid( std::string(TEST_DATA_DIR) + "/ellipsoid_0.ply");
+  Mesh ellipsoid_warped( std::string(TEST_DATA_DIR) + "/ellipsoid_warped.ply");
+  Mesh::MeshType inMesh = ellipsoid.get_poly_data();
```

Comment:
One example of many in this pull request. Please use Mesh outside of a function, and if it's needed the functions themselves can call Mesh::getVTKMesh to get it.